blueflood-elasticsearch/src/main/java/com/rackspacecloud/blueflood/io/EventElasticSearchIO.java

```diff
@@ -54,49 +58,89 @@ public EventElasticSearchIO(ElasticClientManager manager) {
     }
 
     @Override
-    public void insert(String tenant, List<Map<String, Object>> events) throws Exception {
+    public void insert(String tenantId, Map<String, Object> event) throws Exception {
         final Timer.Context eventInsertTimerContext = eventInsertTimer.time();
-        BulkRequestBuilder bulk = client.prepareBulk();
-
-        for (Map<String, Object> event : events) {
-            event.put(Event.FieldLabels.tenantId.toString(), tenant);
-            IndexRequestBuilder requestBuilder = client.prepareIndex(EVENT_INDEX, ES_TYPE)
-                    .setSource(event)
-                    .setRouting(tenant);
-            bulk.add(requestBuilder);
-        }
-        bulk.execute().actionGet();
+        event.put(Event.FieldLabels.tenantId.toString(), tenantId);
+
+        elasticsearchRestHelper.indexEvent(event);
         eventInsertTimerContext.stop();
     }
 
+//    @Override
+//    public void insert(String tenant, List<Map<String, Object>> events) throws Exception {
+//        final Timer.Context eventInsertTimerContext = eventInsertTimer.time();
+//        BulkRequestBuilder bulk = client.prepareBulk();
+//
+//        for (Map<String, Object> event : events) {
+//            event.put(Event.FieldLabels.tenantId.toString(), tenant);
+//            IndexRequestBuilder requestBuilder = client.prepareIndex(EVENT_INDEX, ES_TYPE)
+//                    .setSource(event)
+//                    .setRouting(tenant);
+//            bulk.add(requestBuilder);
+//        }
+//        bulk.execute().actionGet();
+//        eventInsertTimerContext.stop();
+//    }
+
     @Override
     public List<Map<String, Object>> search(String tenant, Map<String, List<String>> query) throws Exception {
         final Timer.Context eventSearchTimerContext = eventSearchTimer.time();
-        BoolQueryBuilder qb = boolQuery()
-                .must(termQuery(Event.FieldLabels.tenantId.toString(), tenant));
+        String result = elasticsearchRestHelper.fetchEvents(tenant, query);
+        eventSearchTimerContext.stop();
```

Comment:
This `stop` should probably be in the `finally` of a `try` around the `fetchEvents` call.

Reply:
Didn't notice that. There was no try catch block earlier. So, looks like it skip through my mind. Now, I added try/catch/finally in all such methods.